flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -1146,6 +1150,54 @@ void CodeGenAction::embedOffloadObjects() {
   }
 }
 
+void CodeGenAction::linkBuiltinBCLibs() {
```

Comment:
This is a fairly large chunk of code - does it come from Clang in some way - if so, is it possible to have one common function shared, rather than two copies?

Reply:
+1

Reply:
It does come from clang but it is distributed across a few classes with a lot more options, so there was not simple way to simply re-use the code. I don't know if we want to bring in all that code since it would add a lot more complexity, and I don't know if we need this in the future or not.